assets/js/blocks/product-gallery/inner-blocks/product-gallery-large-image/frontend.tsx

```diff
@@ -78,5 +91,59 @@ interactivityStore(
 				},
 			},
 		},
+		effects: {
+			woocommerce: {
+				scrollInto: ( store: Store ) => {
+					// Scroll to the selected image with a smooth animation.
+					if (
+						store.selectors?.woocommerce?.isSelected( store ) &&
+						store.context.woocommerce.isDialogOpen ===
+							isDialogStatusChanged
+					) {
+						store.ref.scrollIntoView( {
+							behavior: 'smooth',
+							block: 'nearest',
+							inline: 'center',
+						} );
+					}
+
+					// Scroll to the selected image when the dialog is being opened without an animation.
+					if (
+						store.context.woocommerce.isDialogOpen &&
+						store.context.woocommerce.isDialogOpen !==
+							isDialogStatusChanged &&
+						store.selectors?.woocommerce?.isSelected( store ) &&
+						store.ref.tagName === 'IMG' &&
```

Comment:
This is me being very nit-picky; I believe we should avoid targeting the `tagName` as it is too broad (and potentially can cause conflicts).

For example, I have changed it to `classList.contains` in https://github.com/woocommerce/woocommerce-blocks/pull/11087 to avoid the Next/Previous buttons triggering the Dialog: https://github.com/woocommerce/woocommerce-blocks/pull/11087/files

Reply:
Thanks for the suggestion, but I don't think that it is an approach that we should follow (I will add a specific comment in #11087).

I realized that the check on the tag IMG isn't necessary. So, I just removed it :)